4231 notification settings

eshark9312 opened the pull request:
Resolves #4231 

Email address and its verification states are assumed to be in membership context, 
    which are expected to be resolved from #4232.
![image](https://user-images.githubusercontent.com/129978066/233082419-1a4f37ee-18ba-4bd1-be66-e77d8896cae0.png)

Call back for "Save changes" should be implemented properly.
![image](https://user-images.githubusercontent.com/129978066/233082779-e8ec5ba0-48ef-4c4e-8aa2-363315334d72.png)

chrlschwb reviewed (approved):
LGTM

thesan reviewed (changes requested):
This PR restores the language tab which is not in the scope of #4231. There are also a bunch of commented code which should be removed.

Finally:
> Email address and its verification states are assumed to be in membership context

If by "membership context" you mean the react context provided at the root of the app. It's not the right solution. The email address should be stored in local storage not a React context (although it could be provided from there in addition to be in the local storage) but the rest of the information are just queried on the backend.

eshark9312 commented:
Bundles of comments are removed. 
I added a story for the component in this PR, so called as "WarningWithAction".

thesan reviewed (changes requested):
@eshark9312 

Please address the [first review](https://github.com/Joystream/pioneer/pull/4356#pullrequestreview-1392166099).

Also symbols icon should be added in: `packages/ui/src/common/components/icons/symbols/` and show in this story: https://pioneer-2-storybook-git-fork-eshark9312-4231-n-a56af9-joystream.vercel.app/?path=/story/common-activities-icon--default

In addition please make `WarningWithAction` a variation of the `Warning` component. What I mean is if e.g: `<Warning title="..." content={...} button={...} description={...} />` would show the way `<WarningWithAction title="..." button={...} description={...}>{...}</WarningWithAction>` shows. The goal is to limit duplicated component in the app.

eshark9312 commented:
I removed Language tab from Settings page addressing review#1.
Warning component's variation and the symbol icons are included in the corresponding storybook.

chrlschwb commented on packages/ui/src/app/pages/Settings/EmailNotifications.tsx:
```diff
@@ -0,0 +1,99 @@
+
+  const { hasMembers } = useMyMemberships()
+
+  const [subscribed, setSubscribed] = useState(!!email)
```
shouldn't it be, like this?
`  const [subscribed, setSubscribed] = useState(!email)`
Otherwise I'm not getting the empty page.

eshark9312 replied:
When the email field is empty or null then the empty page is shown .
If you set the initial email state for mocking to null then you'd get empty page.
email be queried to the backend.

chrlschwb commented on packages/ui/src/app/pages/Settings/EmailNotifications.tsx:
```diff
@@ -0,0 +1,99 @@
+    !hasMembers
+      ? (alert('here openModal function comes'), //here openModal function comes
+        (inputstr = prompt('Email notifications verify modal', 'default@email.com')), //here email verify modal comes
+        setEmail(inputstr ?? ''))
```
Please find the correct modal for memebership creation/selection and put it here.

eshark9312 replied:
I found the modal and put it here.
Now It works

chrlschwb commented on packages/ui/src/app/pages/Settings/EmailNotifications.tsx:
```diff
@@ -0,0 +1,99 @@
+                  falseLabel={notifiedCheck ? <TextMedium>No</TextMedium> : <TextMedium bold>No</TextMedium>}
+                  checked={notifiedCheck}
+                  onChange={setNotifiedCheck}
+                />
```
default should be false = "no"

matrizia replied:
Yes,you're right. I've fixed the problem

eshark9312 replied:
Yes,you're right. I've fixed the problem

chrlschwb commented on packages/ui/src/app/pages/Settings/components/SaveChangesButton.tsx:
```diff
@@ -0,0 +1,19 @@
+
+const saveChanges = () => {
+  alert('All changes are saved successfully') //Here SaveFunctions comes
+}
```
Please find the relevant email notification API and use it here.

eshark9312 replied:
sorry , I can't find the api.
I think the backend is not ready for email notification yet.

chrlschwb commented on packages/ui/src/app/pages/Settings/EmailNotifications.tsx:
```diff
@@ -0,0 +1,99 @@
+    if (!notifiedCheck) {
+      setSaveBtnEnabled(true)
+      return
+    }
```
include email verification code here

eshark9312 replied:
Email verification should be carried out on backend, shouldn't it?

thesan reviewed (changes requested):
1. Please add a story for the entire page similar to [`@app/pages/Proposals/ProposalPreview.stories.tsx`](https://github.com/Joystream/pioneer/blob/dev/packages/ui/src/app/pages/Proposals/ProposalPreview.stories.tsx) (you shouldn't need to mock as much state if any as in the example). You will need to pull `dev` again to do this.
2. Remove the commented bits of code.
3. Run `yarn lint:fix`
